gui/default/syncthing/core/syncthingController.js

```diff
@@ -581,6 +597,27 @@ angular.module('syncthing.core')
             $http.get(urlbase + '/system/config').success(function (data) {
                 updateLocalConfig(data);
                 console.log("refreshConfig", data);
+
+                //update web addres
+                for (var index in $scope.config.devices) {
+                    var device = $scope.config.devices[index];
+                    var subId = device.deviceID.substr(0, 6)
+                    if (device.webUi) {
+                        if (!(subId in $scope.webAddress)) {
+                            $scope.webAddress[subId] = ""
+                        } 
+                    } else {
+                        delete $scope.webAddress[subId]
+                    }
+                }
+
+                // set web address port
```

Comment:
I think it would be cleaner to use `$scope.findDevice()` directly to get the device config when needed and grab the port from there, rather than having to maintain `webAddressPort` just for this purpose.